autogpt_platform/autogpt_libs/autogpt_libs/api_key/key_manager.py

```diff
@@ -0,0 +1,44 @@
+import secrets
+import string
+import hmac
+
+class APIKeyManager:
```

Comment:
thank you for making changes so quickly. 

I think maybe I didn't explain myself clearly enough on the API key generation.

We want to use token_urlsafe to generate the random key: https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe

then we can use hashlib.sha256 to hash the key (and we never store or keep the raw key again). So for example, this class would become something like this: 


```
import secrets
import hashlib

class APIKeyManager:
    PREFIX = "agpt_"
    PREFIX_LENGTH = 8
    POSTFIX_LENGTH = 8

    def generate_api_key(self):
        """Generate a secure API key with agpt_ prefix."""
        raw_key = f"{self.PREFIX}{secrets.token_urlsafe(32)}"
        prefix = raw_key[:self.PREFIX_LENGTH]
        postfix = raw_key[-self.POSTFIX_LENGTH:]
        hashed_key = hashlib.sha256(raw_key.encode()).hexdigest()
        return raw_key, prefix, postfix, hashed_key

    def verify_api_key(self, provided_key: str, stored_hash: str) -> bool:
        """Verify if a provided API key matches the stored hash."""
        return hashlib.sha256(provided_key.encode()).hexdigest() == stored_hash```